hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CustomFeesValidator.java

```diff
@@ -84,12 +89,14 @@ public Set<CustomFee> validateForCreation(
             final var collector = accountStore.getAccountById(fee.feeCollectorAccountIdOrElse(AccountID.DEFAULT));
             validateTrue(collector != null, INVALID_CUSTOM_FEE_COLLECTOR);
 
+            final var isSpecified = fee.hasFixedFee() || fee.hasFractionalFee() || fee.hasRoyaltyFee();
+            validateTrue(isSpecified, CUSTOM_FEE_NOT_FULLY_SPECIFIED);
+
             switch (fee.fee().kind()) {
                 case FIXED_FEE -> validateFixedFeeForCreation(
                         tokenType, fee, createdToken, tokenRelationStore, tokenStore, fees);
-                case FRACTIONAL_FEE -> validateTrue(
-                        isFungibleCommon(tokenType), CUSTOM_FRACTIONAL_FEE_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON);
-                case ROYALTY_FEE -> validateRoyaltyFeeForCreation(tokenType, fee, tokenRelationStore, tokenStore);
+                case FRACTIONAL_FEE -> validateFractionalFeeForCreation(tokenType, fee, fees);
```

Comment:
I find it surprising that methods called `validateXYZ()` change the value of `fees`. Maybe there is an easy solution to separate the check and the modification of `fees`.

Reply:
Agreed can change name of the function. Since we are adding all custom fees based on validation, I am not sure how it can be separated.